clarity/src/vm/types/mod.rs

```diff
@@ -1397,6 +1397,15 @@ impl From<StandardPrincipalData> for StacksAddress {
     }
 }
 
+impl From<PrincipalData> for StacksAddress {
```

Comment:
Please remove this conversion. It should never be possible to convert a smart contract address into a standard address.

Reply:
Done.